Code/Framework/AzCore/Tests/Math/PlaneTests.cpp

```diff
@@ -47,7 +47,13 @@ namespace UnitTest
     TEST(MATH_Plane, TestSet)
     {
         Plane pl;
+#if AZ_DEBUG_BUILD
+        AZ_TEST_START_TRACE_SUPPRESSION;
+#endif
```

Comment:
I guess `AZ_DEBUG_BUILD` is needed in these cases because `AZ_MATH_ASSERT` only happens in debug builds.

My suggestion is that the same way math has its own macro for assert, then math tests environment can define its own trace suppression macro.

````
#if AZ_DEBUG_BUILD
    #define AZ_MATH_TEST_START_TRACE_SUPPRESSION     AZ_TEST_START_TRACE_SUPPRESSION
    #define AZ_MATH_TEST_STOP_TRACE_SUPPRESSION(x)   AZ_TEST_STOP_TRACE_SUPPRESSION(x)
#else
    #define AZ_MATH_TEST_START_TRACE_SUPPRESSION
    #define AZ_MATH_TEST_STOP_TRACE_SUPPRESSION(x)
#endif
````

Then all the places inside tests would just be using `AZ_MATH_TEST_START_TRACE_SUPPRESSION` and `AZ_MATH_TEST_STOP_TRACE_SUPPRESSION`, which will be much cleaner.

If in the future `AZ_MATH_ASSERT` changes and it also triggers for profile builds, it'll be much easier of a change to make the macros match in the tests.

Reply:
sure, thanks, put this into its own header since its used by two files